esphome/components/i2c/__init__.py

```diff
@@ -80,8 +83,9 @@ async def to_code(config):
         cg.add(var.set_scl_pullup_enabled(config[CONF_SCL_PULLUP_ENABLED]))
 
     cg.add(var.set_frequency(int(config[CONF_FREQUENCY])))
-    cg.add(var.set_timeout(int(config[CONF_TIMEOUT].total_microseconds)))
     cg.add(var.set_scan(config[CONF_SCAN]))
+    if CORE.using_esp_idf:
```

Comment:
The simplest here is to do this which opens it up if we allow the timeout to be set on arduino later without changing this line again.
```suggestion
    if CONF_TIMEOUT in config:
```